### PR TITLE
Feature/SCORING-1 Question Bank Automation 1.1

### PR DESCRIPTION
**Pull request by oliverrsebastian**
#45  1. Create Question Bank

### CODE REVIEW

File: src/test/java/com/future/function/qa/steps/Scoring/QuestionBank/QuestionBankSteps.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import java.util.Collections;
+import net.thucydides.core.annotations.Steps;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class QuestionBankSteps extends BaseSteps {
```
Comment (jonathan016) [inline src/test/java/com/future/function/qa/steps/Scoring/QuestionBank/QuestionBankSteps.java:21]: Rename method parameter variable names for each method in this class

File: src/test/resources/features/QuestionBank/QuestionBank.feature [hunk 2]
```diff
@@ -0,0 +1,31 @@
+@QuestionBank @Regression
+Feature: Question Bank
+
+  Background:
+    Given user prepare question bank request
+
+  @Negative @QuestionBank
+  Scenario: Create question bank without logging in
+    And user prepare auth request
```
Comment (jonathan016) [inline src/test/resources/features/QuestionBank/QuestionBank.feature:9]: Move `And user prepare auth request` to `Background`

File: src/test/resources/features/QuestionBank/QuestionBank.feature [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    Then question bank response code should be 401
+
+  @Negative @QuestionBank
+  Scenario: Create question bank with empty title and description after logging in as admin
+    And user prepare auth request
+    When user do login with email "admin@admin.com" and password "administratorfunctionapp"
+    And user hit create question bank endpoint with title "" and description ""
+    Then question bank response code should be 400
+    And question bank error response has key "title" and value "NotBlank"
+    And question bank error response has key "description" and value "NotBlank"
```
Comment (jonathan016) [inline src/test/resources/features/QuestionBank/QuestionBank.feature:21]: Add `And user hit logout endpoint` after this line (think as teardown) for this scenario, so the login data is not persisted in other scenarios (due to use of autowired).

Some feature files still do not use this, but in future tests will be corrected

File: src/test/resources/features/QuestionBank/QuestionBank.feature [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+  @Positive @QuestionBank
+  Scenario: Create question bank after logging in as admin
+    And user prepare auth request
+    When user do login with email "admin@admin.com" and password "administratorfunctionapp"
+    And user hit create question bank endpoint with title "QA Title 1" and description "QA Description 1"
+    Then question bank response code should be 201
+    And question bank response data has title "QA Title 1"
+    And question bank response data has description "QA Description 1"
+    And question bank response data has id that should not be blank or null
```
Comment (jonathan016) [inline src/test/resources/features/QuestionBank/QuestionBank.feature:31]: Add `And user hit logout endpoint` after this line (think as teardown) for this scenario, so the login data is not persisted in other scenarios (due to use of autowired).

Some feature files still do not use this, but in future tests will be corrected

File: src/main/java/com/future/function/qa/api/question_bank/QuestionBankAPI.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    base = super.prepare()
+        .basePath(Path.QUESTION_BANK);
+    return base;
+  }
+
+  @Step
+  public Response createQuestionBank(QuestionBankWebRequest request, Cookie cookie) {
+
+    return doByCookiePresent(cookie,
+        () -> createWithCookie(request, cookie),
```
Comment (jonathan016) [inline src/main/java/com/future/function/qa/api/question_bank/QuestionBankAPI.java:28]: Return `Supplier<Response>` as other API classes do

**Review by jonathan016 — CHANGES_REQUESTED** [review]

**Comment by jonathan016** [comment]
Any update on this? cc @oliverrsebastian

**Review by karnandohuang — APPROVED** [review]

**Comment by oliverrsebastian** [comment]
SOLVED on PR #98

**Pull request closed without merging**